packages/react/src/auto/hooks/useRelationshipInputController.tsx

```diff
@@ -69,32 +71,85 @@ const useRelatedModelManager = (props: {
   return { path, metadata, modelManager };
 };
 
-const useGetSelectedRelatedModelRecords = (props: { modelManager: AnyModelManager; fieldProps: ControllerRenderProps<any, string> }) => {
-  const { modelManager, fieldProps } = props;
-  const [selectedRecordId, setSelectedRecordId] = useState<string | undefined>(fieldProps.value);
+const useGetSelectedRelatedModelRecords = (props: {
+  modelManager: AnyModelManager;
+  fieldProps: ControllerRenderProps<any, string>;
+  metadata: FieldMetadataFragment;
+}) => {
+  const { findBy } = useAutoFormMetadata();
+  const { modelManager, fieldProps, metadata } = props;
+
+  const currentRecordId = typeof findBy === "string" ? findBy : undefined;
+  const isBelongsToField = metadata.configuration.fieldType === FieldType.BelongsTo;
+  const hasInitialBelongsToSelection = currentRecordId && fieldProps.value;
+  const inverseFieldApiId = (metadata.configuration as GadgetHasOneConfig | GadgetHasManyConfig).inverseField?.apiIdentifier;
 
-  const [{ data: selectedRecord, fetching: fetchingSelectedError, error: fetchSelectedRecordError }] = useFindOne(
-    modelManager as any,
-    selectedRecordId ?? "",
-    {
-      pause: !selectedRecordId, // Only fetch with a selected records
-    }
+  const [selectedRecordIds, setSelectedRecordIds] = useState<string[]>(
+    isBelongsToField && hasInitialBelongsToSelection ? [fieldProps.value] : []
   );
 
+  // These are needed for HasOne/HasMany fields. To show that a record on the other model is deselected, we need to excluded it from this query
+  const [deselectedRecordIds, setDeselectedRecordIds] = useState<string[]>([]);
```

Comment:
Can you use the form state for this instead of setStates? It seems like this is going to be annoying to keep in sync with the form state, and would not update if someone did a manual set into the form state from somewhere else

Reply:
This `deselectedRecordIds` useState is only used as an excluder filter on the the useFindMany call that looks up the selected related model records, because without it, there would be no way to deselect related records on hasOne/hasMany fields. 
This info can't be stored on the current field's path for the form state without disrupting how the field values are converted into GQL requests. I suppose this could be stored on a separate form path, but that is effectively the same as doing a useState with unused extra functionality

Reply:
This approach is no longer around